agent/src/main/java/com/teamscale/jacoco/agent/commit_resolution/git_properties/GitPropertiesLocatorUtils.java

```diff
@@ -160,25 +175,70 @@ public static ProjectRevision getProjectRevisionFromGitProperties(
 	/** Returns a pair of the zipfile entry name and parsed properties, or null if no git.properties were found. */
 	public static Pair<String, Properties> findGitPropertiesInFile(
 			File file, boolean isJarFile) throws IOException {
-		if (isJarFile) {
-			return findGitPropertiesInJarFile(file);
+		if (isNestedArchivePath(file.getPath())) {
+			return findGitPropertiesInNestedArchiveFile(file);
+		} else if (isJarFile) {
+			return findGitPropertiesInArchiveFile(file);
 		}
 		return findGitPropertiesInDirectoryFile(file);
 	}
 
-	private static Pair<String, Properties> findGitPropertiesInJarFile(File file) throws IOException {
+	private static boolean isNestedArchivePath(String filePath) {
+		return (filePath.contains(WAR_FILE_ENDING) && filePath.endsWith(JAR_FILE_ENDING))
+				|| (filePath.contains(JAR_FILE_ENDING) && filePath.indexOf(
+				JAR_FILE_ENDING) != filePath.length() - JAR_FILE_ENDING.length());
```

Comment:
I thought that in case of a fat jar, the case of having multiple .jar occurrences could be relevant and then this checks that the first .jar occurrence is not at the end. Do you think, it's worth handling that case or should I ignore it. Then I can just remove the second condition after the `||`

Reply:
Ok makes sense. Could you add an inline comment with that as explanation?

Reply:
I split it up into two small methods. I think the names explain it now.